testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -505,6 +507,35 @@ def test_fasta2select_ClustalW(self, tmpdir):
         assert len(
             sel['mobile']) == 23090, "selection string has unexpected length"
 
+    def test_fasta2select_nogap(self, tmpdir):
+        """test align.fasta2select() on aligned FASTA with no gap (Issue #3124)"""
+        try:
+            reference = mda.Universe(self.ref).select_atoms('all')
+            target = mda.Universe(self.tgt).select_atoms('all')
+            # Rename histidine and remove alt conformation to permit alignment
+            for residue_reference in reference.residues:
+                if residue_reference.resname in ['HIP', 'HIE', 'HID']:
+                    residue_reference.resname = 'HIS'
+            for residue_target in target.residues:
+                if residue_target.resname in ['HIP', 'HIE', 'HID']:
+                    residue_target.resname = 'HIS'
+            with tmpdir.as_cwd():
+                alignment = align.sequence_alignment(target.select_atoms('protein and not altloc B'),
+                                                     reference.select_atoms('protein and not altloc B'))
+                fasta = '>ref\n{}\n>mobile\n{}'.format(alignment[0], alignment[1])
+                file_fasta = open('alignment.aln', "w+")
+                file_fasta.write(fasta)
+                file_fasta.close()
+                ref_resids = [a.resid for a in
+                              reference.select_atoms('protein').select_atoms('name CA and not altloc B')]
+                target_resids = [a.resid for a in
+                                 target.select_atoms('protein').select_atoms('name CA and not altloc B')]
+                sel = align.fasta2select('alignment.aln',
+                                         is_aligned=True,
+                                         ref_resids=ref_resids,
+                                         target_resids=target_resids)
+        except UnboundLocalError:
```

Comment:
In my opinion it's best to not do this within a try/except block.

InsteadI suggest you run the test and check that the outputs of fasta2select are reasonable. If there is an issue with that code path, then the test will highlight it without having to capture the error.